docs/DeveloperGuide.md

```diff
@@ -301,15 +301,113 @@ A possible drawback of this uniform design is that it may not be the most approp
+
+**After having considered all the pros & cons, we have decided to implement this feature so that the user can have the best possible User Experience!**
+
+----
+
+**Allow users to use mouse to click on tabs**
+
+_This feature is one of the most debated feature in our team._ 
+
+We are well aware of the module's constraints that the app must be optimized for CLI. 
+
+Yet, 5 todo-apps have been studied by our team (macOS's reminders, Microsoft To Do, Todoist, OmniFocus) and we found all of these apps only have one way to change tabs, and it is by clicking on them. 
+Although forcing users to use CLI to change tab can be easily implemented, it will make the app unintuitive to most users (except CLI guru), and hence **cannot attract new users** to change to our new app from their old apps(of which they are so used to clicking on tabs). 
+
+Also, even for CLI guru, typing something like `switch CS2030` to switch to a new tab while it can be done with just a single click is not something we think they will do. 
+
+One last reason is that we think it's very convenient for users to use mouse to click on various different tabs in during a short period to just get a sense of upcoming tasks & lessons and related contacts to help them. This "fast jump between tabs" cannot be done by CLI.
+
+----
+
+**Edit/Delete contacts/lessons/tasks by Index**
+
+|  | Pros | Cons |
+| ---- | ----- | ------- |
+| **Option 1 (current choice):** Use Index to edit/delete | Very "intuitive" for users, they can use the index being displayed to edit/delete the entity they need to | Cannot delete contacts from the upcoming tab. There is no concrete Id for tasks/lessons/contacts, theirs Ids (in this case: index) depends on their position in the current tab|
+| **Option2:** Assign each contact/lesson/task a concrete Id | Each entity is associated with only one Id, can edit/delete anywhere in the app | Force users to remember long sequences of Ids and hence create a mental burden everytime they want to edit/delete an entity |
+
+**We have chosen the 1st option to make users feel comfortable using our app**
+
+----
+**Unified command syntax**
+
+We have put in a lot of effort to make the command syntax unified, means that all commands have very similar syntax. They can be observed from the following tables:
+
+| Command | Format |
+| -- | -------- |
+| add module | `M add m/CODE n/NAME` | 
+| add lesson | `L add m/CODE t/TYPE d/DATE a/ADDRESS` | 
+| add task | `T add n/NAME d/DATE [m/CODE] [r/REMARK]` | 
+| add contact | `C add n/NAME p/PHONE_NUMBER e/EMAIL [t/TAG]...` | 
+
+| Command | Format |
+| -- | -------- |
+| edit module | `M edit CODE [m/NEW_CODE] [n/NAME]` | 
+| edit lesson | `L edit INDEX [m/CODE] [t/TYPE] [d/DATE] [a/ADDRESS]` | 
+| edit task | `T edit INDEX [n/NAME] [d/DATE] [m/CODE] [r/REMARK]` | 
+| edit contact | `C edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [t/TAG]...` |
+
```

Comment:
Hmm, should this be under code design considerations? Not sure if command syntax is a feature?

Reply:
it is a feature right? It doesn't affect the back-end